src/main/java/backend/resource/TurboLabel.java

```diff
@@ -55,26 +62,40 @@ public static TurboLabel exclusive(String repoId, String group, String name) {
      */
     public TurboLabel(TurboLabel label) {
         this.actualName = label.getActualName();
+        this.grouping = initGroupMembership();
+        this.simpleName = initSimpleName();
+        this.groupName = initGroupName();
         this.colour = label.getColour();
         this.repoId = label.getRepoId();
     }
 
     public TurboLabel(String repoId, Label label) {
         this.actualName = label.getName();
+        this.grouping = initGroupMembership();
+        this.simpleName = initSimpleName();
+        this.groupName = initGroupName();
         this.colour = label.getColor();
         this.repoId = repoId;
     }
 
     public TurboLabel(String repoId, SerializableLabel label) {
         this.actualName = label.getActualName();
+        this.grouping = initGroupMembership();
+        this.simpleName = initSimpleName();
+        this.groupName = initGroupName();
```

Comment:
But those are final fields which can only be initialized in a constructor. I will just reuse the basic constructor for the other more complex constructors.